core/shared/platform/common/posix/posix_blocking_op.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2023 Midokura Japan KK.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "platform_api_extension.h"
+
+#ifdef OS_ENABLE_WAKEUP_BLOCKING_OP
```

Comment:
Not whether we should make the feature configurable, e.g. `cmake -DWAMR_BUILD_WAKEUP_BLOCKING_OP=1/0`?

Reply:
are you suggesting to make it a build-time configurable?
i myself do not have a strong opinion either ways.

Reply:
Yes, build-time configurable. How about adding `cmake -DWAMR_BUILD_WAKEUP_BLOCKING_OP=1/0`, and by default it is enabled? My concern is that some one may want to disable it if not needed, and it can reduce binary (though little).

And if added, had better update doc/build_warm.md accordingly.

Reply:
ok.
in that case, i want to make it similar to hw_bound_check.
do you know why WASM_DISABLE_HW_BOUND_CHECK uses a negative notation? ("disable" rather than "enable")

Reply:
Because the hw bound check for some 64-bit platforms is enabled by default, using `cmake -DWAMR_BUILD_HW_BOUND_CHECK=1/0` makes a little confusing, we used `cmake -DWAMR_DISABLE_HW_BOUND_CHECK=1` instead, and then the macro `WASM_DISABLE_HW_BOUND_CHECK` was used accordingly. Maybe it is not so consistent with other cmake variables and compiler macros :)

Reply:
so, how about:

* make it enabled by default where available
* introduce `-DWAMR_DISABLE_WAKEUP_BLOCKING_OP=1/0`
* define `OS_ENABLE_WAKEUP_BLOCKING_OP` only when enabled

Reply:
It is ok for me, looks well.

Reply:
done